src/fast_library.cpp

```diff
@@ -1284,3 +1284,7 @@ bool store_data_to_stats_server(unsigned short int graphite_port, std::string gr
         return false;
     }
 }
+
+uint32_t convert_hex_as_string_to_uint(std::string hex) {
+    return std::strtoul(hex.c_str(), 0, 16);
```

Comment:
Thank you for patch! I started review. 

It's very C-ish way to this task and it requires return value checks.

I can suggest typical for C++ approach which uses streams:
```unsigned int x;   
std::stringstream ss;
ss << std::hex << "fffefffe";
ss >> x;
```

After value extraction attempt (ss >> x), you can check any error using:
```
if (ss.fail()) {
    // Return error
}
```

stringstream object set this flag when any parser errors happened.

Reply:
It affects my programming in C, but not in C ++.

I see a solution in using std::pair and return a value with error flag like this.

    ss << std::hex << hex_as_string;
    ss >> x;

    // return the value and error flag
    return std::make_pair(x, ss.fail());


But I do not like this solution.
How best to return an error from this function?

Reply:
You can add additional variable. std::paid syntax is quite complex for it.

You can try something like:
```

bool convert_hex_as_string_to_uint(std::string hex, uint32_t& value) {
    value = 1234;
    return true;
}
```